frontend/app/components/Spots/SpotPlayer/components/AccessModal.tsx

```diff
@@ -35,70 +36,64 @@ function AccessModal() {
     spotStore.pubKey ? `?pub_key=${spotStore.pubKey.value}` : ''
   }`;
 
+  useEffect(() => {
+    if (spotStore.pubKey) {
+      const interval = Object.keys(expirationValues).find(key => expirationValues[key as Intervals] === spotStore.pubKey.expiration);
+      if (interval) setSelectedInterval(Number(interval) as Intervals);
+    }
+  }, [spotStore.pubKey]);
+
   const menuItems = [
     {
-      key: Intervals.hour,
+      key: Intervals.hour.toString(),
       label: <div>One Hour</div>,
     },
     {
-      key: Intervals.threeHours,
+      key: Intervals.threeHours.toString(),
       label: <div>Three Hours</div>,
     },
     {
-      key: Intervals.day,
+      key: Intervals.day.toString(),
       label: <div>One Day</div>,
     },
     {
-      key: Intervals.week,
+      key: Intervals.week.toString(),
       label: <div>One Week</div>,
     },
   ];
-  const onMenuClick = ({ key }: { key: Intervals }) => {
-    const val = expirationValues[key];
-    if (
-      spotStore.pubKey?.expiration &&
-      Math.abs(spotStore.pubKey?.expiration - val) / val < 0.1
-    ) {
-      return;
-    }
-    void spotStore.generateKey(spotId, val);
+
+  const onMenuClick = async (info: { key: string }) => {
+    console.log('Menu item clicked:', info.key);
+    const val = expirationValues[Number(info.key) as Intervals];
+    setSelectedInterval(Number(info.key) as Intervals);
+    setLoadingKey(true);
+    console.log('Generating key with expiration:', val);
+    await spotStore.generateKey(spotId, val);
+    setLoadingKey(false);
   };
 
   const changeAccess = async (toPublic: boolean) => {
     if (isPublic && !toPublic && spotStore.pubKey) {
-      if (
-        await confirm({
-          header: 'Confirm',
-          confirmButton: 'Disable',
-          confirmation:
-            'Are you sure you want to disable public sharing for this spot?',
-        })
-      ) {
-        void spotStore.generateKey(spotId, 0);
-      }
+      await spotStore.generateKey(spotId, 0);
```

Comment:
this means that we will not ask when user want to change it and backend has to do extra work if someone just wanted to click around

Reply:
How often does users change it? once or twice. the modal confirmation is too much for a little action.